### ISSUE TITLE
[Bug] Creating SmPLMethodCFG causes error if CtExecutable contains for loop

### ISSUE THREAD

**Issue by J-Merle** [INRIA/spoon#4716]
Trying to create a new SmPLMethodCFG causes an error if the CtExecutable contains a for loop

Here is a minimal code sample to reproduce

Input:
```java
public class FailingProcessor extends AbstractProcessor<CtExecutable<?>> {
    @Override
    public void process(CtExecutable<?> element) {
        SmPLMethodCFG cfg = new SmPLMethodCFG(element);
    }

    private void forLoop() {
        for (int i : Arrays.asList(1, 2, 3, 4)) {
            System.out.println(i);
        }
    }

    public static void main(String[] args) {
        Launcher launcher = new Launcher();
        launcher.addInputResource(new FileSystemFile(new File("./src/main/java/FailingProcessor.java"))); // Change the path to match the current file path
        launcher.addProcessor(new FailingProcessor());
        launcher.run();
    }
}

```

Output:
```java
Exception in thread "main" java.lang.ClassCastException: class spoon.support.reflect.code.CtForEachImpl cannot be cast to class spoon.reflect.code.CtIf (spoon.support.reflect.code.CtForEachImpl and spoon.reflect.code.CtIf are in unnamed module of loader 'app')
	at spoon.smpl.SmPLMethodCFG.<init>(SmPLMethodCFG.java:341)
	at processors.FailingProcessor.process(FailingProcessor.java:15)
	at processors.FailingProcessor.process(FailingProcessor.java:12)
	at spoon.support.visitor.ProcessingVisitor.scan(ProcessingVisitor.java:72)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:184)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:149)
	at spoon.reflect.visitor.CtScanner.visitCtClass(CtScanner.java:357)
	at spoon.support.reflect.declaration.CtClassImpl.accept(CtClassImpl.java:58)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:194)
	at spoon.support.visitor.ProcessingVisitor.scan(ProcessingVisitor.java:68)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:184)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:149)
	at spoon.reflect.visitor.CtScanner.visitCtPackage(CtScanner.java:677)
	at spoon.support.reflect.declaration.CtPackageImpl.accept(CtPackageImpl.java:88)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:194)
	at spoon.support.visitor.ProcessingVisitor.scan(ProcessingVisitor.java:68)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:184)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:149)
	at spoon.reflect.visitor.CtScanner.visitCtPackage(CtScanner.java:676)
	at spoon.support.reflect.declaration.CtPackageImpl.accept(CtPackageImpl.java:88)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:194)
	at spoon.support.visitor.ProcessingVisitor.scan(ProcessingVisitor.java:68)
	at spoon.support.QueueProcessingManager.process(QueueProcessingManager.java:118)
	at spoon.support.QueueProcessingManager.process(QueueProcessingManager.java:132)
	at spoon.support.compiler.jdt.JDTBasedSpoonCompiler.process(JDTBasedSpoonCompiler.java:195)
	at spoon.Launcher.process(Launcher.java:790)
	at spoon.Launcher.run(Launcher.java:734)
	at processors.FailingProcessor.main(FailingProcessor.java:28)

Process finished with exit code 1

```

* OS: Windows 10
* JDK: 11
* Spoon version: 10.1.1

**Comment by monperrus** [comment]
cc/ @mkforsb WDYT?

**Comment by mkforsb** [comment]
The really short answer is that there are several Java elements that are not supported by spoon-smpl at this time. None of the loop elements are currently supported, and adding support would require considerable effort.

However, an exception trace is obviously not a reasonable way of communicating that some things aren't supported. I've submitted a PR in which I've added the for-loop elements to a mechanism that filters out unsupported elements, while also logging warning messages for every filtered element. This should be a slight improvement on the current user experience.
